File: zikzak_inappwebview_platform_interface/lib/src/context_menu/context_menu_settings.dart
import 'package:zikzak_inappwebview_internal_annotations/zikzak_inappwebview_internal_annotations.dart';

import 'context_menu.dart';

part 'context_menu_settings.g.dart';

///Class that represents available settings used by [ContextMenu].
@ExchangeableObject(copyMethod: true)
class ContextMenuSettings_ {
  ///Whether all the default system context menu items should be hidden or not. The default value is `false`.
  bool hideDefaultSystemContextMenuItems;

  ContextMenuSettings_({this.hideDefaultSystemContextMenuItems = false});
}

File: zikzak_inappwebview_macos/macos/zikzak_inappwebview_macos/Sources/zikzak_inappwebview_macos/InAppWebView/ContextMenuSettings.swift
//
//  ContextMenuSettings.swift
//  zikzak_inappwebview
//
//  Ported from iOS to support macOS custom context menus.
//  https://github.com/arrrrny/zikzak_inappwebview/issues/196
//

import Foundation

public class ContextMenuSettings: ISettings<NSObject> {

    /// Whether all the default system context menu items should be hidden or not.
    /// When `true`, only the custom [ContextMenu] menu items are shown.
    /// The default value is `false`.
    var hideDefaultSystemContextMenuItems = false

    override init() {
        super.init()
    }
}
